### PR TITLE
Todo List

### PR DESCRIPTION
**Pull request by Donmark2k**
### Highlights 🥇

**To do list`**

**Learning objectives**

- Use webpack to bundle JavaScript.
- Learn how to use proper ES6 syntax.
- Understand improvements that ES6 brings to JavaScript.
- Use npm as software packages system.

**What I Have Done:**

- Added linters. The [JavaScript + HTML + CSS linter](https://github.com/microverseinc/linters-config/tree/master/html-css-js).
- I use let and const in a correct way.
- I display the current date below the navigation bar.
- I set up a new project with webpack that is based on the webpack exercise I have already completed.
- I write a function to iterate over the tasks array and populate an HTML list item element for each task.
- On load, the page render dynamically  in the dedicated placeholder.
-  I add event listener to the checkbox (change).
- I update items object's value for completed key upon user actions.
- I implement a function for the "Clear all completed" button (use filter() method).

**Followed Rules:**

- Made sure [no linter errors](https://github.com/microverseinc/linters-config)
- Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
- Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- Followed our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### CODE REVIEW

File: src/modules/status.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          } else if ((e.target.id === `input${task.index}`) && (task.completed === true)) {
+            task.completed = false;
+            e.target.checked = false;
+            localStorage.setItem('array', JSON.stringify(array));
+          }
+        });
+      });
+    });
+  });
+}
```
Comment (AmaduKamara) [inline src/modules/status.js:22]: - [ ] When I click on a task to mark it as completed, it does not happen in an instance, not until I refresh the page and then click completed again before it reflects in the localStorage. All todos I mark as completed reflects in the localStorage but when I refresh the browser, the changes maintain in the localStorage but the ticks on the check boxes for completed items do not show. Kindly watch the video below for a better demonstration and please try to fix it so it is interactive upon user interaction with the app. 👍🏽

[Interactive Requirement Video Demo](https://www.loom.com/share/ba794cdecbeb4a83a47c707eedd8bcc2)

File: src/template.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <script src="https://kit.fontawesome.com/70657c6fc0.js" crossorigin="anonymous"></script>
     <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,-50..200" />
-    <link rel="stylesheet" href="style.css">
+    <!-- <link rel="stylesheet" href="style.css"> -->
```
Comment (AmaduKamara) [inline src/template.html]: - [ ] Kindly remove commented code from your codebase so your code looks clean and professional as it's a good convention to have a good clean code

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @Donmark2k,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Good descriptive PR 👍🏽
✅ Good commit messages 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/modules/status.js (hunk 1 above)
Comment (Donmark2k) [reply src/modules/status.js:22]: I solved this by assigning the checkbox to empty string and it worked

File: src/template.html (hunk 2 above)
Comment (DeliceLydia) [reply src/template.html]: addressed

File: src/modules/status.js (hunk 1 above)
Comment (DeliceLydia) [reply src/modules/status.js:22]: Addressed

**Review by DeliceLydia — APPROVED** [review]
## Hi @Donmark2k, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**